app/models/hyrax/file_metadata.rb

```diff
@@ -1,6 +1,24 @@
 # frozen_string_literal: true
 
 module Hyrax
+  ##
+  # Casts a resource to an associated FileMetadata
+  #
+  # @param [Valkyrie::StorageAdapter::File] file
+  #
+  # @return [Hyrax::FileMetadata]
+  # @raise [ArgumentError]
+  def self.FileMetadata(file)
+    raise ArgumentError if file.is_a?(Valkyrie::Resource) # maybe heavy handed?
```

Comment:
what sort of error message could we put here? what kind of `Valkyrie::Resources` might be accidentally passed here?

Reply:
the skys the limit?

if we're asking questions like this, i think i'd rather just drop the line and let the query fail in whatever way it fails.

Reply:
ah, i take it back. the existing test covering this provides a good example:

if you pass in, e.g. a FileSet, you'd get a FileMetadata with an unfortunate reference to the FileSet. this seems like a common enough pitfall to guard against; it would result in unwanted metadata resources otherwise.